arbnode/sync_monitor.go

```diff
@@ -0,0 +1,131 @@
+package arbnode
+
+import (
+	"sync/atomic"
+
+	"github.com/offchainlabs/nitro/arbutil"
+	flag "github.com/spf13/pflag"
+)
+
+type SyncMonitor struct {
+	config      *SyncMonitorConfig
+	inboxReader *InboxReader
+	txStreamer  *TransactionStreamer
+	coordinator *SeqCoordinator
+	initialized bool
+}
+
+func NewSyncMonitor(config *SyncMonitorConfig) *SyncMonitor {
+	return &SyncMonitor{
+		config: config,
+	}
+}
+
+type SyncMonitorConfig struct {
+	BlockBuildLag               uint64 `koanf:"block-build-lag"`
+	BlockBuildSequencerInboxLag uint64 `koanf:"block-build-sequencer-inbox-lag"`
+	CoordinatorMsgLag           uint64 `koanf:"coordinator-msg-lag"`
+}
+
+var DefaultSyncMonitorConfig = SyncMonitorConfig{
+	BlockBuildLag:               20,
+	BlockBuildSequencerInboxLag: 0,
+	CoordinatorMsgLag:           200,
```

Comment:
Why is `CoordinatorMsgLag` so much larger than `BlockBuildLag`? Shouldn't they be the same (or even CoordinatorMsgLag maybe should be a bit smaller as we have stricter requirements on our sequencers?)